### DISCUSSION TITLE
Authenticated layouts with CatchBoundary/ErrorBoundary

### DISCUSSION THREAD

**Discussion by gregbty** [remix-run/remix#1201] [Q&A] [answered]
I have a scenario that I can't seem to find a great solution for when using `CatchBoundary` and `ErrorBoundary`. If a user is logged in and an error occurs i.e. missing route or unexpected, ideally, I'd like to be able to display the navigation menu (including the user section). Currently I am not able to do that via the `root` boundaries because the loader data is inaccessible, which includes the user's login metadata. Is there some mechanism to handle this in Remix outside of moving and duplicating the layout in other nested layouts and creating nested boundaries?

**Comment by sergiodxa** [comment, marked as answer]
You can create a layout route and put the catch boundary and error boundary there, then in the root return the authenticated user data and render the header there or only return the user data and call useMatches from the catch/error boundaries to get the route data and render the header.

**Reply by gregbty** [reply]
That's what I'm currently doing but it requires to duplicating the logic in each top level layout file. However, it seems like [Flat Layout Routes](https://remix.run/docs/en/v1/api/conventions#flat-layout-route) may work for this scenario. I'd just need to nest all of my other routes underneath a directory starting with `__`.

**Reply by gregbty** [reply]
Actually I misread. Your second suggestion works for my case as well and is a bit simpler than using the Flat Layout Routes. `useMatches` gives me the data I need. Thanks!
